apps/mobile/src/features/send-refactor/send-flow-provider.tsx

```diff
@@ -0,0 +1,104 @@
+import { ReactNode, createContext, useContext, useReducer } from 'react';
```

Comment:
Nice work here 👍  I like the `useReducer` pattern. It would be good to add some tests to cover state being set correctly

Reply:
I'd love to, but apparently vitest refuses to even test hooks. Did you get a chance at looking into making it work with RN?

Reply:
You can do it with `vi.mock` I think. Like how we mocked `useSelector` in `stacks-keychains.read.spec.ts` :
```
vi.mock('react-redux', () => ({
  useSelector: vi.fn(),
}));
```

We can re-visit this in the cooldown, I added a note [here](https://linear.app/leather-io/issue/LEA-2079)